test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,436 @@
+        assertTrue(currentReadCount1 > currentReadCount2);
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(60000);
+        log.info("Scaling down controller instances from 3 to 2");
+
+        currentWriteCount2 = eventData.get();
+        currentReadCount2 = eventReadCount.get();
+
+        log.info("Read count after controller failover after sleep  {}", currentReadCount2);
+        log.info("Write count after controller failover after sleep  {}", currentWriteCount2);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount2 > currentWriteCount1);
+        //ensure reads are happening
+        assertTrue(currentReadCount2 > currentReadCount1);
+
+        //Scale down SSS, controller to 1 instance each.
+        segmentStoreInstance.scaleService(1, true);
+        controllerInstance.scaleService(1, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(new Random().nextInt(50000) + 60000);
+        log.info("Scaling down  to 1 controller, 1 SSS instance");
+
+        currentWriteCount1 = eventData.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Stop write flag status {}", stopWriteFlag);
+        log.info("Stop read flag status {}", stopReadFlag);
+        log.info("Event data {}", eventData.get());
+        log.info("Read count {}", eventReadCount.get());
+        log.info("Read count with SSS and controller failover after sleep {}", currentReadCount1);
+        log.info("Write count with SSS and controller failover after sleep {} ", currentWriteCount1);
+    }
+
+    private CompletableFuture<Void> startWritingIntoTxn(final AtomicLong data, final EventStreamWriter<Long> writer,
+                                                 final AtomicBoolean stopWriteFlag) {
+        return CompletableFuture.runAsync(() -> {
+            while (!stopWriteFlag.get()) {
+                    try {
+                        Transaction<Long> transaction = writer.beginTxn(5000, 3600000, 29000);
+                        //each transaction has 100 events
+                        for (int j = 0; j < 100; j++) {
+                            long value = data.incrementAndGet();
+                            transaction.writeEvent(String.valueOf(value), value);
+                            log.debug("Writing event {} into transaction {}", value, transaction);
+                        }
+                        //commit Txn
+                        transaction.commit();
+                        log.debug("Transaction {} committed", transaction);
+                        log.debug("Transaction status {}", transaction.checkStatus());
+                    } catch (Throwable e) {
+                        log.warn("Exception while writing events in the transaction : {}", e);
```

Comment:
Log the transaction id for which transaction failed, also we need to distinguish between transaction creation failure v/s a transaction commit() failure. If there is a failure during Txn.commit() it will cause a mismatch in `event written count` and `events read count`.